### ISSUE TITLE
network interface view does not wrap status on mobile

### ISSUE THREAD

**Issue by jelly** [cockpit-project/cockpit#19190]
### Explain what happens

The `span` part of this rule no longer applies as it is now a `div`
```
#network-interface {
  @at-root {
    #network-interface-name {
      font-weight: var(--pf-v5-c-card__title--FontWeight);
    }

    .network-interface-status > span {
      overflow-wrap: anywhere;
    }
  }
}
```

The issue can be seen here:

![image](https://github.com/cockpit-project/cockpit/assets/67428/218e32e6-1a76-4453-b78f-67f444d7de38)

We should break on words and show the IPv4 and IPv6 address underneath each other.

### Where is the problem in Cockpit?

None

### Server operating system

None

**Comment by garrett** [comment]
Status wraps here in Firefox.

One problem we have is that the header isn't wrapping. Perhaps that's what you're seeing?

![image](https://github.com/cockpit-project/cockpit/assets/10246/dac47411-2a0f-429d-b2d0-2b45e14dd6d9)

**Comment by jelly** [comment]
> Status wraps here in Firefox.
> 
> One problem we have is that the header isn't wrapping. Perhaps that's what you're seeing?
> 
> ![image](https://user-images.githubusercontent.com/10246/259695673-dac47411-2a0f-429d-b2d0-2b45e14dd6d9.png)

Interesting! Our pixel tests run on Chromium so is this simply a browser issue?

**Comment by garrett** [comment]
Having the text be a grid, especially with `grid-auto-flow: column;` breaks the layout for devices with longer names (which most of mine have).

If we wanted it to be in one line when possible and wrap otherwise, that should be a flex instead, probably with `display: flex; flex-flow: wrap;` instead.

(This is all on `.network-interface-details-title` in `networking.scss`)

**Comment by garrett** [comment]
> Interesting! Our pixel tests run on Chromium so is this simply a browser issue?

No, not an issue. This has to do with wrapping or filling the space. If there's more space, it won't wrap. The title is too large and overflows in both browsers. How they handle the rest might be due to it being unspecified.

> Either way the css seems unrequired?

This particular CSS might not be required here, or it might be. It depends. It could even be the case that PF might've changed the wrapping behavior too. (They have for tables at some point, for example. They might've for description lists as well.)

**Comment by garrett** [comment]
We'd probably also want the card title to look more like this too:

![image](https://github.com/cockpit-project/cockpit/assets/10246/242e2bf7-9523-4b23-ae3c-b7bb3086bca5)
(not wrapping, when the name is short)

![image](https://github.com/cockpit-project/cockpit/assets/10246/cf7253bd-1e85-47c1-b1f8-86fd7666cb75)
(automatically wrapping, when the name is long)

It would mean removing grid related stuff in the class and adding flex related CSS instead, such as:

```scss
.network-interface-details-title {
  display: flex;
  flex-flow: wrap;
  align-items: baseline;
  gap: var(--pf-v5-global--spacer--xs) var(--pf-v5-global--spacer--md);
}
```

And sizing `network-interface-hw` and `network-interface-mac` to be standard size.
